core/src/main/java/com/datastax/oss/driver/internal/core/ssl/SniSslEngineFactory.java

```diff
@@ -29,8 +31,8 @@
 public class SniSslEngineFactory implements SslEngineFactory {
```

Comment:
One question to make it clear. This `SniSslEngineFactory` is created only once per lifetime of the application, right?
Or is it created once per session? If once per session then where is that reusability of `fakePorts`?
And `newSslEngine()` is called only at the start of the app as many times as there are endpoints in the cloud that we are connecting to, yes?
Once it will be done there are no more `newSslEngine` calls?

Reply:
I think SSL session caching is done at the `SSLContext` level (need to confirm that).

We have one `SniSslEngineFactory` per driver session, and each factory uses its own `SSLContext` (created in `CloudConfigFactory` actually). So I don't think there can be any collision across driver sessions.

Reply:
Just to confirm, the cache is indeed [local to the SSLContext](https://github.com/JetBrains/jdk8u_jdk/blob/master/src/share/classes/sun/security/ssl/SSLContextImpl.java#L47).

Reply:
Ok, thx for checking this.